docs/src/userguide/config/event_config.rst

```diff
@@ -126,13 +126,176 @@ Example configuration for a custom event input that is not supported by Klio:
+
+    .. code-block:: yaml
+
+        name: my-cool-job
+        pipeline_options:
+          streaming: True
+        job_config:
+        events:
+            inputs:
+            - type: bq
+              columns:
+              - entity_id
+              dataset: bigquery_dataset
+              project: gcp_project
+              table: bigquery_table
 
 
+Example configuration for `Google BigQuery`_, by specifying a query in the BigQuery dialect:
+
+    .. code-block:: yaml
+
+        name: my-cool-job
+        pipeline_options:
+          streaming: True
+        job_config:
+        events:
+            inputs:
+            - type: bq
+              query:
+                SELECT * FROM [gcp_project.bigquery_dataset.bigquery_table]
+
+.. option:: job_config.events.inputs[].type
+
+    Value: ``bq``
+
+    | **Runner**: Dataflow, Direct
+    | *Required*
+
+.. option:: job_config.events.inputs[].columns[]
+
+    Not applicable if query is specified.
+
+    A list of strings specifying the columns to read events from.
+    If only one column is provided, the value will be returned as a bytestring.
+    If no columns are specified, or if more than one column is specified,
+    the results including the column names will be serialized to a JSON bytestring,
+    e.g. ``'{"field1": "foo", "field2": bar"}'``.
+
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+.. option:: job_config.events.inputs[].columns[].<column> STR
+
+    A column in the table or query result used to build the event input from.
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+.. option:: job_config.events.inputs[].table STR
+
+    Name of the event input table.
+
+    If specified all data of the table will be used as input of the current source.
+    The ID must contain only letters ``a-z``, ``A-Z``, numbers ``0-9``,
+    or underscores ``_``.
+
+    If dataset and query arguments are not specified,  then the table argument must
+    contain the entire table reference specified as:
+    ``'DATASET.TABLE'`` or ``'PROJECT:DATASET.TABLE'``.
+
+    | **Runner**: Dataflow, Direct
+    | *Required unless query provided*
+
+.. option:: job_config.events.inputs[].dataset STR
+
+    Name of the event input table's dataset.
+
+    Ignored if the table reference is fully specified by the table argument or
+    if a query is specified.
+
+    | **Runner**: Dataflow, Direct
+    | *Required if query not provided and table reference does not include dataset*
+
+.. option:: job_config.events.inputs[].project STR
+
+    Name of the event input table's project.
+
+    Ignored if the table reference is fully specified by the table argument or
+    if a query is specified.
+
+    | **Runner**: Dataflow, Direct
+    | *Required if query is not provided and table reference does not include project*
+
+.. option:: job_config.events.inputs[].query STR
+
+    Query string supplying the columns. Mutually exclusive with specifying the
+    table field.
```

Comment:
Similarly, can you code format ``table``? And isn't it also mutually exclusive with `project` and `dataset`, too?